Sources/Apple/AppleAuthenticator.swift

```diff
@@ -34,49 +33,44 @@ public final class AppleAuthenticator: NSObject {
 extension AppleAuthenticator: Authenticator {
   /// SignIn user
   ///
-  /// Will return promise with the `Response` object on success or with `Error` on error.
-  public func signIn(from presentingViewController: UIViewController) -> Promise<Response> {
-    let promise = Promise<Response>()
-    processingPromise = promise
-    appleSignIn(on: presentingViewController, with: nil)
-    return promise
+  /// Will asynchronously return the `Response` object on success or `Error` on error.
+  public func signIn(from presentingViewController: UIViewController) async throws -> Response {
+    try await appleSignIn(on: presentingViewController, with: nil)
   }
-  
+
   /// SignIn user with `nonce` value
   ///
   /// Nonce is usually needed when doing auth with an external auth provider (e.g. firebase).
-  /// Will return promise with the `Response` object on success or with `Error` on error.
-  public func signIn(from presentingViewController: UIViewController, with nonce: Nonce) -> Promise<Response> {
-    let promise = Promise<Response>()
-    processingPromise = promise
-    appleSignIn(on: presentingViewController, with: nonce)
-    return promise
+  /// Will asynchronously return the `Response` object on success or `Error` on error.
+  public func signIn(from presentingViewController: UIViewController, with nonce: Nonce) async throws -> Response {
+    try await appleSignIn(on: presentingViewController, with: nonce)
   }
-  
+
   /// Clears the signIn footprint and logs out the user immediatelly.
   public func signOut() {
     storage.removeObject(forKey: storageUserIdKey)
-    rejectSignIn(with: .cancelled)
+    storage.setValue(false, forKey: storageAuthenticatedKey)
+    continuation = nil
   }
   
   /// Returns the current authentication state.
   public var isAuthenticated: Authenticated {
     storage.string(forKey: storageUserIdKey) != nil && storage.bool(forKey: storageAuthenticatedKey)
   }
   
-  /// Checks the current auth state and returns the boolean value as promise.
-  public var checkAuthentication: Promise<Authenticated> {
+  /// Checks the current auth state and returns the boolean value asynchronously.
+  public func checkAuthentication() async -> Authenticated {
     guard let userId = storage.string(forKey: storageUserIdKey) else {
-      return .value(false)
+      return false
     }
-    
-    return Promise { seal in
+
+    return await withCheckedContinuation { continuation in
       ASAuthorizationAppleIDProvider().getCredentialState(forUserID: userId) { credentialsState, _ in
```

Comment:
Maybe we can consider using the [async version](https://developer.apple.com/documentation/authenticationservices/asauthorizationappleidprovider/3175423-getcredentialstate#discussion) if this method.

Reply:
Wow, didn't notice the concurrency method. Yes, let's do it. cc @yllfejziu

Reply:
Nice catch 👍